src/inmanta/moduletool.py

```diff
@@ -1751,13 +1751,32 @@ def _move_data_files_into_namespace_package_dir(self, build_path: str) -> None:
         """
         python_pkg_dir: str = os.path.join(build_path, "inmanta_plugins", self._module.name)
         model_dir: str = os.path.join(python_pkg_dir, "model")
-        if os.path.exists(model_dir):
-            raise ModuleBuildFailedError(
-                msg="There is already a model directory in %s. "
-                "The `inmanta_plugins.%s.model` package is reserved for bundling the inmanta model files. "
-                "Please use a different name for this Python package."
-                % (os.path.join(self._module.path, "inmanta_plugins", self._module.name), self._module.name)
-            )
+        files_dir: str = os.path.join(python_pkg_dir, "files")
+        templates_dir: str = os.path.join(python_pkg_dir, "templates")
+        for problematic_dir_path in [model_dir, files_dir, templates_dir]:
+            if os.path.exists(problematic_dir_path):
+                problematic_dir_name = os.path.basename(problematic_dir_path)
```

Comment:
The general flow here feels a bit off to me. You start with a name for each of these, then build the paths for each seperately, loop over them, and use `os` to get the names you started with.

And as an added bonus, if you start out with `Collection[tuple[<dir>, <description>]]` you don't need the if-else with exhaustiveness check either.